src/Data/Decimal.purs

```diff
@@ -0,0 +1,190 @@
+module Data.Decimal
+  ( Decimal()
+  , decimal
+  , getMantissa
+  , getExponent
+  , fromInt
+  , prettyDecimal
+  ) where
+
+import Prelude
+import Data.Array as A
+import Data.Either as E
+import Data.Maybe as M
+import Data.Ord as O
+import Data.String as S
+import Control.Monad.Rec.Class as Rec
+
+import Test.StrongCheck as SC
+import Test.StrongCheck.Gen as Gen
+
+type DecimalR =
+  { mantissa :: Int
+  , exponent :: Int
+  }
+
+-- | An abstract type for exact computing with decimal numbers of finite expansion.
+newtype Decimal = Decimal DecimalR
+
+instance showDecimal :: Show Decimal where
+  show (Decimal d) =
```

Comment:
`Show` is kinda supposed to print PureScript code which can be used to reconstruct the value, so maybe there's a better option that that too, since you can't construct decimals with the record can you?